### ISSUE TITLE
Allow multi-line recipe signatures

### ISSUE THREAD

**Issue by casey** [casey/just#1266]
This came up in #1265. We can do this without requiring a continuation of any kind, just ignore newlines when parsing recipe parameters.

Great first issue!

**Comment by neunenak** [comment]
Hey @casey , I thought I'd take a crack at this. https://github.com/casey/just/pull/1272 seems to work with at least one test justfile I've made - I'm still figuring out how the just codebase works, how tests work, etc., and I'm not sure if this is the best way to solve this parsing problem, so I don't want to call this ready for review just yet. I'd appreciate it if you had any suggestions about what to tell the parser to do here.

**Comment by casey** [comment]
Nice!

I think the difficulty with the approach in #1272 is that we want to ignore newlines in grammatical sub constructs. So here:

```
    loop {
        if self.next_is(Identifier) || self.next_is(Dollar) {
            positional.push(self.parse_parameter(ParameterKind::Singular)?);
        } else if self.next_is(Eol) {
            self.accept(Eol)?;
        } else {
            break;
        }
    }
```

We also want to ignore newlines in parameters, which would require a change to parse-parameter.

One potential solution is to switch the parser between eol-sensitive and non eol-sensitive mode. In non EOL sensitive modes, parser functions that checked for tokens/returned tokens would switch the TokenKind of line endings whitespace.

I'm not really sure which approach is better, it really depends on how gross the code is for each.

**Comment by jraygauthier** [comment]
One of the main pain point for me.

**Comment by laniakea64** [comment]
@jraygauthier Why is it a main pain point to use [\ or parentheses](https://just.systems/man/en/chapter_47.html#outside-recipe-bodies)?

**Comment by jraygauthier** [comment]
@laniakea64: Lol, because I wasn't aware this has been fixed yet (as it appears in v1.15) and was using v1.13:

```bash
$ just build-debug-features
error: Unknown start of token:
   |
33 | build-debug-features: \
   |                       ^

$ just --version
just 1.13.0
```

I will upgrade a more recent > v1.15 version.

Now, to me this ticket should be marked as fixed by PR https://github.com/casey/just/pull/1551, right? It might be a good idea to update Q/A https://github.com/casey/just/discussions/1265 as well.

**Comment by casey** [comment]
Yup, I think this can be closed.
